src/scripts/main.js

```diff
@@ -1,3 +1,375 @@
+});
+
+function generateIndex(numberList) {
+  return Math.floor(Math.random() * numberList.length);
+};
+
+function createNewNumb() {
+  const freeCell = [...allCellList].filter(item => {
+    return item.dataset.currNumb === '';
+  });
+
+  const numbValue = firstNumb[generateIndex(firstNumb)];
+  const cellGlobalIdx = generateIndex(freeCell);
+  const cell = freeCell[cellGlobalIdx];
+
+  cell.classList.add(`field-cell--${numbValue}`);
+  cell.innerHTML = numbValue;
+  cell.dataset.currNumb = numbValue;
+}
+
+function letsStart(e) {
+  startMessage.classList.add('hidden');
+  createNewNumb();
+  createNewNumb();
+
+  e.target.blur();
+  startButton.addEventListener('click', reset);
+  startButton.removeEventListener('click', letsStart);
+  startButton.classList.remove('start');
+  startButton.classList.add('restart');
+  startButton.innerHTML = 'Restart';
+  document.addEventListener('keydown', moveCell);
+}
+
+function reset(e) {
+  [...allCellList].forEach(cell => {
+    cell.dataset.currNumb = '';
+    cell.className = `field-cell`;
+    cell.innerHTML = ``;
+  });
+
+  createNewNumb();
+  createNewNumb();
+  e.target.blur();
+  globalScoreElement.innerHTML = 0;
+  loseMessage.classList.add('hidden');
+  winMessage.classList.add('hidden');
+}
+
+function showLocalScore(score) {
+  const localScoreElem = document.createElement('span');
+  const scoreTimeDur = '1s';
+
+  document.querySelector('.info').append(localScoreElem);
+  localScoreElem.className = 'local-score';
+  localScoreElem.innerHTML = '+' + score;
+  localScoreElem.style.transitionDuration = scoreTimeDur;
+
+  setTimeout(() => {
+    localScoreElem.style.bottom = '15px';
+    localScoreElem.style.opacity = 0;
+  }, 200);
+
+  setTimeout(() => {
+    localScoreElem.remove();
+  }, parseInt(scoreTimeDur) * 1000);
+}
+
+function performCellTransform(currCell, newCell, e) {
+  const activeCell = document.createElement('div');
+  let isMerge = false;
+
+  activeCell.style.left = currCell.offsetLeft + 'px';
+  activeCell.style.top = currCell.offsetTop + 'px';
+  activeCell.style.width = currCell.clientWidth + 'px';
+  activeCell.style.height = currCell.clientHeight + 'px';
+
+  activeCell.className = `
+    active-cell
+    field-cell
+    field-cell--${currCell.innerText}
+  `;
+
+  activeCell.innerHTML = currCell.innerText;
+  activeCell.dataset.currNumb = currCell.dataset.currNumb;
+  activeCell.style.transitionDuration = cellTransDur + 'ms';
+  supportField.append(activeCell);
+  currCell.className = 'field-cell';
+  currCell.innerHTML = '';
+  currCell.dataset.currNumb = '';
+
+  if (e.keyCode === 37 || e.keyCode === 39) {
+    activeCell.style.left = newCell.offsetLeft + 'px';
+  }
+
+  if (e.keyCode === 38 || e.keyCode === 40) {
+    activeCell.style.top = newCell.offsetTop + 'px';
+  }
+
+  if (newCell.dataset.currNumb === activeCell.dataset.currNumb) {
+    newCell.style.transitionDuration = cellTransDur + 'ms';
+    newCell.dataset.currNumb *= 2;
```

Comment:
Why 2? Try to avoid magic numbers. All numbers that are unclear from the context should be put into constants with a name that is understandable in terms of content